refactor(ivy): remove calls into other instructions from pipe instructions

crisbeto opened the pull request:
A while ago we made a pass through all instructions to make sure that none of them call directly into other instructions, however it seems like missed the `pipeBind*` since they still call into the pure functions. The result is that we have some unnecessary duplicated accesses of global state like `getLView` which are called twice in a row with nothing changing.

These changes move the common functionality into a shared file and make the pipe instructions call into them with the global state instead.

pkozlowski-opensource reviewed (approved):
LGTM

kara commented on packages/core/src/render3/instructions/shared.ts:
```diff
@@ -1850,3 +1851,138 @@ export function renderInitialStyling(
+ * @param thisArg Optional calling context of pureFn
+ * @returns Updated or cached value
+ */
+export function pureFunction1Internal(
```
The `shared` file is already pretty huge. Any reason we can't put these functions into `pure_function.ts`?

crisbeto replied:
There's no reason, I thought that the convention for functions used by multiple instructions was to put them in `shared.ts`.

crisbeto commented:
@kara what would be the next step here?

kara commented:
@crisbeto My understanding was that my feedback hadn't been addressed, so I was waiting on that

crisbeto commented:
The feedback has been addressed @kara.

crisbeto committed:
refactor(ivy): remove calls into other instructions from pipe instructions

A while ago we made a pass through all instructions to make sure that none of them call directly into other instructions, however it seems like missed the `pipeBind*` since they still call into the pure functions. The result is that we have some unnecessary duplicated accesses of global state like `getLView` which are called twice in a row with nothing changing.

These changes move the common functionality into a shared file and make the pipe instructions call into them with the global state instead.

kara reviewed (approved):
LGTM

kara commented:
[presubmit](http://test/OCL:288738131:BASE:288738217:1578511413943:8773335f)
